src/System.Net.Http/src/System/Net/Http/Unix/CurlResponseHeaderReader.cs

```diff
@@ -0,0 +1,282 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace System.Net.Http
+{
+    internal struct CurlResponseHeaderReader
+    {
+        private const string HttpPrefix = "HTTP/";
+
+        private readonly HeaderBufferSpan _span;
+
+        public CurlResponseHeaderReader(IntPtr buffer, ulong size)
+        {
+            Debug.Assert(buffer != IntPtr.Zero);
+            // In actuality, size will be <= CURL_MAX_HTTP_HEADER (1024*100) bytes.
```

Comment:
I added an assert for `CURL_MAX_HTTP_HEADER` in the the callback method. I was going to add it here, but decided it made more sense to define `CURL_MAX_HTTP_HEADER` with the other interop definitions and didn't want `CurlResponseHeaderReader` to depend on the interop because `CurlResponseHeaderReader` is tested in the `System.Net.Http` unit tests that don't currently depend on any interop file and I didn't want to have to pull the interop files into the unit tests just for this.

I left the `int.MaxValue` assert here because that's really all we care about here.